save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/WarnPluginConfig.kt

```diff
@@ -165,5 +169,29 @@ data class WarnPluginConfig(
          * ```[WARN] /path/to/resources/ClassNameTest.java:2:4: Class name in incorrect case```
          */
         internal val defaultOutputPattern = Regex("(.+):(\\d*):(\\d*): (.+)")
+        internal val defaultExtraConfigPattern = Regex("// RUN: (.+)")
+    }
+}
+
+/**
+ * @property args1 arguments to be inserted *before* file name
+ * @property args2 arguments to be inserted *after* file name
+ */
+data class ExtraFlags(
+    val args1: String,
+    val args2: String,
+) {
+    companion object {
+        const val KEY_ARGS_1 = "args1"
+        const val KEY_ARGS_2 = "args2"
+
+        /**
+         * Construct [ExtraFlags] from provided map
+         *
+         * @param map a map possibly containing values for [args1] and [args2], denoted by keys [KEY_ARGS_1] and [KEY_ARGS_2]
+         * @return [ExtraFlags]
+         */
+        fun from(map: Map<String, String>) =
```

Comment:
I guess that from should be done from initial string. For encapsulation.
But actually I would also create a deserializer 🤦‍♂️(Kidding)

Reply:
I tried to refactor it, yes, your variant looks better, because extractFrom has too many arguments and too long functions for constructor